### PR TITLE
Implement Mail service

### PR DESCRIPTION
**Pull request by BillyZac**
Implements a mail service on bookit-server. This will be used to send email notifications to Bookit users. For example, when a Wipro employee is invited to use Bookit, or when a booking fails.

Test the mail service by making a GET request to `/testSendMail`. Then check Bruce's inbox to see the message.

### CODE REVIEW

File: src/model/Mail.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    ],
+    body: {
+      content: string,
+      contentType: 'html'
+    }
+  };
+}
+
+
+// {
```
Comment (defpearlpilot) [inline src/model/Mail.ts]: Can we kill this dead comment?

File: src/rest/test_routes.ts [hunk 2]
```diff
@@ -11,4 +12,18 @@ export function configureTestRoutes(app: Express) {
     res.send('test succeeded');
   });
 
+  app.get('/testSendMail', (req: Request, res: Response) => {
```
Comment (defpearlpilot) [inline src/rest/test_routes.ts]: Should probably kill this if we aren't going to expose it

File: src/services/mail/MailService.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export interface MailService {
+  sendMail(
+    senderEmail: string,
+    recipientEmail: string,
+    messageType: string,
+  ): Promise<any>;
```
Comment (defpearlpilot) [inline src/services/mail/MailService.ts]: This is nitpicking but if we don't expect a result, set it to return Promise<void>.

File: src/services/mail/MockMailService.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import {MailService} from './MailService';
+
+export class MockMailService implements MailService {
+  constructor() {
+    logger.info('MockMailService: initializing');
+  }
+
+
+  sendMail(senderEmail: string, recipientEmail: string, messageType: string): Promise<any> {
```
Comment (defpearlpilot) [inline src/services/mail/MockMailService.ts]: As per the above, do we want it to promise a string or nothing at all?

Comment (BillyZac) [inline src/services/mail/MockMailService.ts]: @defpearlpilot Made those changes. Except I left the 'testSendMail' route in place. It's kinda dumb, but it's the only way to hand test the functionality at the moment, I think. We definitely want to remove it soon, though.

**Pull request merged**